tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,1060 @@
+   "source": [
+    "### Example 3: Multiple outputs\n",
+    "Similarly, `TensorDictModule` not only supports multiple inputs but also multiple outputs. To make a `TensorDictModule` instance write to multiple output values values, one must register them in the `out_keys` keyword argument of the constructor.\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "0b7f709b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MultiHeadLinear(nn.Module):\n",
+    "    def __init__(self, in_1, out_1, out_2):\n",
+    "        super().__init__()\n",
+    "        self.linear_1 = nn.Linear(in_1, out_1)\n",
+    "        self.linear_2 = nn.Linear(in_1, out_2)\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        return self.linear_1(x), self.linear_2(x)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "1b2b465f",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "TensorDict(\n",
+       "    fields={\n",
+       "        a: Tensor(torch.Size([5, 3]), dtype=torch.float32),\n",
+       "        output_1: Tensor(torch.Size([5, 4]), dtype=torch.float32),\n",
+       "        output_2: Tensor(torch.Size([5, 10]), dtype=torch.float32)},\n",
+       "    batch_size=torch.Size([5]),\n",
+       "    device=cpu,\n",
+       "    is_shared=False)"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "tensordict = TensorDict({\"a\": torch.randn(5, 3)}, batch_size=[5])\n",
+    "\n",
+    "splitlinear = TensorDictModule(\n",
+    "    MultiHeadLinear(3, 4, 10),\n",
+    "    in_keys=[\"a\"],\n",
+    "    out_keys=[\"output_1\", \"output_2\"],\n",
+    ")\n",
+    "splitlinear(tensordict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "859630c3",
+   "metadata": {},
+   "source": [
+    "As we shown previously, the `TensorDictModule` can take any `nn.Module` and perform the operations on a `TensorDict`. When having multiple input keys and output keys, make sure they match the order in the module.\n",
+    "`TensorDictModule` can work with `TensorDict` instances that contain more tensors than what the `in_keys` attribute indicates. Unless a `vmap` operator is used, the `TensorDict` is modified in-place."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11d2d2a7-6a55-4f31-972b-041be387f9df",
+   "metadata": {},
+   "source": [
+    "### Example 4: Combining multiples `TensorDictModule` with `TensorDictSequence`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89b157d5-322c-45d6-bec9-20440b78a2bf",
+   "metadata": {},
+   "source": [
+    "To combine multiples `TensorDictModule` instances, we can use `TensorDictSequence`. We create a list where each `TensorDictModule` must be executed sequentially. `TensorDictSequence` will take the output of the n-1th `TensorDictModule` in the list and feed it to the nth `TensorDictModule`.\n",
```

Comment:
> `TensorDictSequence` will take the output of the n-1th `TensorDictModule` in the list and feed it to the nth `TensorDictModule`.

this is not true and actually it is important to say that it isn't. You can perfectly have

```python
moda = TensorDictModule(
    Linear(3, 4),
    in_keys=["a"],
    out_keys=["b"],
)
modb = TensorDictModule(
    Linear(3, 4),
    in_keys=["c"],
    out_keys=["d"],
)

seq = TensorDictSequence(moda, modb)
```

which contradicts the sentence above. 

Maybe we can put a similar example to show this. It's important because we **want** `TensorDictSequence` to support cases where not all modules have congruent output / input lists

Reply:
Also I would point that in_keys and out_keys of TensorDictSequence have the right values

Reply:
But the tensordict fed to modb has been affected by moda no? Even if the tensors that modb use arent affected by moda.  it will have keys a, b and c

Reply:
I see what you mean. What about "TDSeq will read and write keys to the tensordict following the sequence of modules provided." Or similar if you find a better phrasing